### PR TITLE
Refactor message page

### PR DESCRIPTION
**Pull request by Pinkdesu**
- Refactored message page

### CODE REVIEW

File: website/frontend/src/tests/mocks/handlers/helpers.ts [hunk 1]
```diff
@@ -22,11 +22,17 @@ export const getApiResponse = <T>(data: T): RPCResponse<T> => ({
...
+    entry: null,
+    value: '',
     source: messageId === MESSAGE_ID_1 ? PROGRAM_ID_1 : PROGRAM_ID_2,
-    destination: '0xd43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d',
     payload: '0x4e6f7420656e6f7567682067617320666f7220696e697469616c206d656d6f72792068616e646c696e67',
-    replyError: '1',
+    exitCode: null,
     timestamp: '2022-06-29T09:54:04.000Z',
+    expiration: null,
+    replyError: '1',
```
Comment (osipov-mit) [inline website/frontend/src/tests/mocks/handlers/helpers.ts:32]: `replyError` doesn't exist in response from now on

File: website/frontend/src/types/message.ts [hunk 2]
```diff
@@ -1,10 +1,18 @@
...
-  source: string;
   id: string;
+  entry: 'Init' | 'Handle' | 'Reply' | null;
+  value: string;
+  source: string;
+  payload: string;
+  exitCode: number | null;
   timestamp: string;
+  expiration: null | number;
   replyError: any;
```
Comment (osipov-mit) [inline website/frontend/src/types/message.ts:12]: as above

**Review by osipov-mit — CHANGES_REQUESTED** [review]

File: website/frontend/src/components/blocks/MessagesList/MessagesList.tsx [hunk 3]
```diff
@@ -1,71 +1,21 @@
...
+  messages: MessageModel[];
 };
 
-export const MessagesList: VFC<Props> = ({ messages }) => {
-  const alert = useAlert();
+const MessagesList = ({ messages }: Props) => (
+  <div>
+    <MessagesListHeader />
+    <ul>
+      {messages.map((message: MessageModel) => (
```
Comment (aderrod94) [inline website/frontend/src/components/blocks/MessagesList/MessagesList.tsx:14]: Better to use getMessages function that return result of the map

File: website/frontend/src/components/blocks/MessagesList/children/MessageItem/MessageItem.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  const alert = useAlert();
+
+  const { id: messageId, exitCode, timestamp } = message;
+
+  const handleCopy = () => copyToClipboard(messageId, alert, 'Message ID copied');
+
+  const status = exitCode ? IndicatorStatus.Error : IndicatorStatus.Success;
+
+  return (
+    <li className={clsx(styles.messageItem, exitCode ? styles.error : styles.success)}>
```
Comment (aderrod94) [inline website/frontend/src/components/blocks/MessagesList/children/MessageItem/MessageItem.tsx:28]: Store classes in the variable and use here

File: website/frontend/src/components/common/CircleIndicator/CircleIndicator.tsx [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+import styles from './CircleIndicator.module.scss';
+import { IndicatorStatus } from './types';
+
+type Props = {
+  status: IndicatorStatus;
+  className: string;
+};
+
+const CircleIndicator = ({ status, className }: Props) => (
+  <span className={clsx(styles.circleIndicator, styles[status], className)} />
```
[the remark this answers is not in the document]
Comment (Pinkdesu) [reply website/frontend/src/components/common/CircleIndicator/CircleIndicator.tsx:12]: I don't see the point in this case.

**Review by aderrod94 — APPROVED** [review]

**Review by osipov-mit — APPROVED** [review]

**Pull request merged**